core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java

```diff
@@ -0,0 +1,155 @@
+import com.netease.arctic.ams.api.CatalogMeta;
+import com.netease.arctic.io.ArcticFileIO;
+import com.netease.arctic.io.ArcticHadoopFileIO;
+import com.netease.arctic.table.ArcticTable;
+import com.netease.arctic.table.BaseUnkeyedTable;
+import com.netease.arctic.table.TableBuilder;
+import com.netease.arctic.table.TableIdentifier;
+import com.netease.arctic.table.TableMetaStore;
+import com.netease.arctic.utils.CatalogUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A wrapper class around {@link Catalog} and implement {@link ArcticCatalog}.
+ */
+public class BaseIcebergCatalog implements ArcticCatalog {
+
+  private CatalogMeta meta;
+  private transient TableMetaStore tableMetaStore;
+  private transient Catalog icebergCatalog;
+
+  @Override
+  public String name() {
+    return meta.getCatalogName();
+  }
+
+  @Override
+  public void initialize(
+      AmsClient client, CatalogMeta meta, Map<String, String> properties) {
+    this.meta = meta;
+    meta.putToCatalogProperties(org.apache.iceberg.CatalogUtil.ICEBERG_CATALOG_TYPE,
+        meta.getCatalogType());
+    tableMetaStore = CatalogUtil.buildMetaStore(meta);
+    icebergCatalog = org.apache.iceberg.CatalogUtil.buildIcebergCatalog(name(),
+        meta.getCatalogProperties(), tableMetaStore.getConfiguration());
+  }
+
+  @Override
+  public List<String> listDatabases() {
+    if (icebergCatalog instanceof SupportsNamespaces) {
+      return tableMetaStore.doAs(() -> ((SupportsNamespaces)icebergCatalog).listNamespaces(Namespace.empty()).stream()
+          .map(namespace -> namespace.level(0)).distinct().collect(Collectors.toList()));
+    } else {
+      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
+       icebergCatalog.getClass().getName()));
+    }
+  }
+
+  @Override
+  public void createDatabase(String databaseName) {
+    if (icebergCatalog instanceof SupportsNamespaces) {
+      tableMetaStore.doAs(() -> {
+        ((SupportsNamespaces)icebergCatalog).createNamespace(Namespace.of(databaseName));
+        return null;
+      });
+    } else {
+      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
+          icebergCatalog.getClass().getName()));
+    }
+  }
+
+  @Override
+  public void dropDatabase(String databaseName) {
+    if (icebergCatalog instanceof SupportsNamespaces) {
+      tableMetaStore.doAs(() -> {
+        ((SupportsNamespaces)icebergCatalog).dropNamespace(Namespace.of(databaseName));
+        return null;
+      });
+    } else {
+      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
```

Comment:
throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",

'do now' should be does not ?